### DISCUSSION TITLE
Server minimum size

### DISCUSSION THREAD

**Discussion by sebco17** [KL-Psychological-Methodology/ESMira#13] [Questions] [answered]
Hi, 
I need to request a hosting service for the server. 
The minimal requirements for ESMira are:

- PHP 7.4 or greater.
- PHP needs write access in the file system (which it usually has).
- For an easy setup: Support for .htaccess files (See below for a setup without .htaccess).

I would like to know if you could also provide a minimal server size requirement.
I imagine that it will depend on the complexity of the studies, the number of studies, the number of participants etc., however, my technical support asked me to provide an overall requirement of the size. Generally, there are no significant problems with increasing later, but it would be useful to understand if there is a minimum or recommended storage, just to understand what size of the server to use at the beginning. Do you have any directions or suggestions?

**Comment by JodliDev** [comment, marked as answer]
In general I would say, that if want to use media items (user images, audio recordings, ...) every now and then, I'd ask for 50 GB of space.
If you plan of relying heavily on media items, I'd ask for at least 300 GB.
Without media items 25 GB (including the operating system) is still plenty.

To give you some specifics:
ESMira itself barely uses up disk space (less than 20 MB). Operating system, web server and PHP will need around 20GB (depending on your setup). But the biggest variable depends on your studies and if your studies use any media item.

The biggest non-media study we have, uses 140 MB. And this one has been running for over a year, has more than 1000 participants and more than 60000 entries in the events file.
Most of our studies never use more than 20 MB of space (small studies usually stay below 1 MB) and at the moment, all our non-media studies from 4 years combined use less than 1 GB of space,

In contrast, we had one study that ran for about two weeks and needed daily pictures of around 50 participants. This study alone needed more than 15 GB of space (which also stems from ESMira not compressing image sizes, which we need to fix at some point. See https://github.com/KL-Psychological-Methodology/ESMira-apps/issues/1).
